tools/LLVMOptimizer.cpp

```diff
@@ -0,0 +1,75 @@
+#include "LLVMOptimizer.h"
+
+using namespace llvm;
+
+llvm::Module *optimize_module(std::string optArgs, llvm::Module *M) {
```

Comment:
this is a lot of code duplication. You need to refactor this preamble into a static function.

Reply:
0 0 Now my current implementation is optimize_function calls optimize_module. Do you think if I should remove optmize_function with only optimize_module remains?